src/install-ocaml-unix.sh

```diff
@@ -6,7 +6,8 @@ set -ex
 # the right version from the system compiler
 
 CURRENT_OCAML=$(opam info -f version ocaml --color=never)
+SWITCHES=$(opam switch list)
 
-if [ "$CURRENT_OCAML" != "$1" ]; then
+if [ "$CURRENT_OCAML" != "$1" ] && [ "$SWITCHES" != *"$1"* ]; then
   opam switch create "$1" "$1"
 fi
```

Comment:
This doesn't mean what I think you want it to (`*` is for globbing, not regex matching). This I think is closer:
```suggestion
if [ "$CURRENT_OCAML" != "$1" ]; then
  case "$SWITCHES" in
    *$1*);;
    *)
      opam switch create "$1" "$1";;
  esac
fi
```
However, it catches sub-strings and breaks with switches containing spaces.